packages/block-library/src/navigation/edit/placeholder/index.js

```diff
@@ -1,94 +1,117 @@
 /**
  * WordPress dependencies
  */
-import { Placeholder, Button } from '@wordpress/components';
+import { Placeholder, Button, Spinner } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { navigation, Icon } from '@wordpress/icons';
+import { speak } from '@wordpress/a11y';
+import { useEffect } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
-
 import useNavigationEntities from '../../use-navigation-entities';
 import PlaceholderPreview from './placeholder-preview';
 import useCreateNavigationMenu from '../use-create-navigation-menu';
 import NavigationMenuSelector from '../navigation-menu-selector';
 
 export default function NavigationPlaceholder( {
+	isSelected,
 	currentMenuId,
 	clientId,
-	onFinish,
-	hasResolvedNavigationMenus,
 	canUserCreateNavigationMenu = false,
+	isResolvingCanUserCreateNavigationMenu,
+	onFinish,
 } ) {
 	const createNavigationMenu = useCreateNavigationMenu( clientId );
 
 	const onFinishMenuCreation = async (
 		blocks,
 		navigationMenuTitle = null
 	) => {
-		if ( ! canUserCreateNavigationMenu ) {
-			return;
-		}
-
 		const navigationMenu = await createNavigationMenu(
 			navigationMenuTitle,
 			blocks
 		);
 		onFinish( navigationMenu, blocks );
 	};
 
-	const { isResolvingPages, isResolvingMenus } = useNavigationEntities();
-
-	const isStillLoading = isResolvingPages || isResolvingMenus;
+	const { isResolvingMenus, hasResolvedMenus } = useNavigationEntities();
 
 	const onCreateEmptyMenu = () => {
 		onFinishMenuCreation( [] );
 	};
 
+	useEffect( () => {
+		if ( ! isSelected ) {
+			return;
+		}
+
+		if ( isResolvingMenus ) {
+			speak(
+				'Loading Navigation block setup placeholder options.',
+				'polite'
+			);
+		}
+
+		if ( hasResolvedMenus ) {
+			speak(
+				'Navigation block setup placeholder options ready.',
+				'polite'
+			);
+		}
+	}, [ isResolvingMenus, isSelected ] );
+
+	const isResolvingActions =
```

Comment:
Only "resolving" in the UI if _both_ of these are resolving. If one is completed then we can show the user something already.

Reply:
This doesn't hide anything, though, right? It only shows or hides the spinner.

Reply:
Basically. It's a low fidelity approach to avoid there being nothing showing on slower connections whilst the resolution states resolve.

If you recall we're going to try and refactor the logic out of `NavigationMenuSelector` into a hook. But in a follow up PR.